apis_core/generic/views.py

```diff
@@ -104,14 +105,35 @@ def get_table_class(self):
 
     def get_table_kwargs(self):
         kwargs = super().get_table_kwargs()
-        columns = self.request.GET.getlist("columns", [])
-        column_fields = [
-            field for field in self.model._meta.fields if field.name in columns
+
+        # we look at the selected columns and exclude
+        # all modelfields that are not part of that list
+        selected_columns = self.request.GET.getlist(
+            "columns",
+            self.get_filterset(self.get_filterset_class()).form["columns"].initial,
+        )
+        modelfields = self.model._meta.fields
+        kwargs["exclude"] = [
+            field.name for field in modelfields if field.name not in selected_columns
         ]
+
+        # now we look at the selected columns and
+        # add all modelfields and annotated fields that
+        # are part of the selected columns to the extra_columns
+        annotationfields = tuple()
```

Comment:
```suggestion
        annotationfields = list()
```
tuple is immutable, no append